### PR TITLE
build: fix owasp check broken

### LINKED ISSUE #4370: Owasp CI is broken
I found that the OWASP check is broken; it doesn't trigger during the change of #4344. It should be triggered when these files are changed.

https://github.com/apache/bookkeeper/blob/c2e8037baa5ad2beaab29c519a5a54a893d00585/.github/workflows/bk-ci.yml#L482-L487

https://github.com/apache/bookkeeper/blob/c2e8037baa5ad2beaab29c519a5a54a893d00585/.github/changes-filter.yaml#L12-L16

### PR DESCRIPTION
**Pull request by shoothzj**
Fix #4370

### CODE REVIEW

**Review by StevenLuMT — APPROVED** [review]
LGTM

File: .github/workflows/bk-ci.yml [hunk 1]
```diff
@@ -505,7 +509,8 @@ jobs:
       - name: Set up JDK 11
         uses: actions/setup-java@v4
         with:
-          java-version: 11
+          distribution: 'temurin'
+          java-version: 21
```
Comment (hangc0276) [inline .github/workflows/bk-ci.yml:513]: Why upgrade the java version to jdk21? All the current CI's java version is jdk11

Comment (shoothzj) [reply .github/workflows/bk-ci.yml:513]: @hangc0276 I previously thought that using JDK 21 for OWASP might be potentially faster and help expose more problems.

Additionally, considering that OpenJDK 11 will reach its end of life in October 2024, do we have plan to upgrade CI to jdk17?

File: .github/changes-filter.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ docs:
   - '**/*.md'
   - '.github/changes-filter.yaml'
   - '.github/ISSUE_TEMPLATE/**'
+  - 'src/owasp-dependency-check-suppressions.xml'
```
Comment (nicoloboschi) [inline .github/changes-filter.yaml:12]: this is wrong, it should go on "need_owasp" section

Comment (shoothzj) [reply .github/changes-filter.yaml:12]: @nicoloboschi It's already in the "need_owasp` section, I think it's also no need to run tests, so I put it in here.

File: .github/workflows/bk-ci.yml (hunk 1 above)
Comment (hangc0276) [reply .github/workflows/bk-ci.yml:513]: At least, we should keep the same with other CI's versions

**Pull request merged**